vllm/model_executor/layers/fla/ops/utils.py

```diff
@@ -0,0 +1,385 @@
+# Copyright (c) 2023-2024, Songlin Yang, Yu Zhang
+
+import functools
+from typing import Optional
+
+import torch
+import triton
+import triton.language as tl
+from packaging import version
+
+
+def contiguous(fn):
+
+    @functools.wraps(fn)
+    def wrapper(ctx, *args, **kwargs):
+        return fn(
+            ctx,
+            *(i if not isinstance(i, torch.Tensor) else i.contiguous()
+              for i in args), **{
+                  k: (v if not isinstance(v, torch.Tensor) else v.contiguous())
+                  for k, v in kwargs.items()
+              })
+
+    return wrapper
```

Comment:
* This seems generally useful enough to put in `vllm/utils.py`
* `contiguous()` is a pretty tough constraint. Do you know why this is needed by so many of the kernels used by RWKV-6?